### PR TITLE
fix(files): hard-cap search result payloads

### PR DESCRIPTION
**Pull request by w3rdist-creator**
## Summary
- cap serialized `search_files` JSON below 64K
- preserve valid JSON and mark size truncation explicitly
- add regression coverage for oversized dense results

## Verification
- `uv run ruff check tools/file_tools.py tests/tools/test_search_output_cap.py`
- `uv run pytest -q tests/tools/test_search_output_cap.py tests/tools/test_tool_result_storage.py` (27 passed)

### CODE REVIEW

File: tools/file_tools.py [hunk 1]
```diff
@@ -1911,6 +1946,7 @@ def search_tool(pattern: str, target: str = "content", path: str = ".",
                 "The results have not changed. Use the information you already have."
```
Comment (teknium1) [inline tools/file_tools.py:1946]: This trims `matches_text` mid-page, but the existing truncated-result hint advances with `offset + limit`. A size-capped response that retained fewer than `limit` complete matches will skip its unseen tail on the next request. Please cap complete source entries before densification and derive the continuation offset from entries actually retained.

**Review by teknium1 — COMMENTED** [review]
Thanks for addressing the unbounded `search_files` result path; current main still densifies and serializes the complete result at `tools/file_tools.py:2113-2135`.

### Problems
- `tools/file_tools.py:1946` truncates the already-densified `matches_text` by character prefix. The normal truncated-result hint advances by `offset + limit` (`tools/file_tools.py:2135-2140` on current main). If the size cap retains fewer than `limit` complete matches, the suggested follow-up offset skips the unseen tail of the original page.
- `tests/tools/test_search_output_cap.py:38-37` checks output size and JSON validity, but not continuation correctness.

### Suggested changes
- Cap complete source matches before densification, then calculate continuation from the number actually retained.
- Add a two-page regression test proving size truncation neither drops nor duplicates matches.

This is an automated hermes-sweeper review.
